client/src/components/AuthPage/Authentication/Authentication.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from "react";
+import { Navigate, Link } from "react-router-dom";
+import {
+  doSignInWithEmailAndPassword,
+  doSignInWithGoogle,
+  doCreateUserWithEmailAndPassword,
+} from "../../../firebase/auth";
+import { useAuth } from "../../../contexts/authContexts";
+import "../../../../node_modules/bootstrap/dist/css/bootstrap.min.css";
+
+const Authentication = () => {
+  const { userLoggedIn } = useAuth; // will use in future to go straight to home page
+
+  const [logInEmail, setLogInEmail] = useState("");
+  const [registerEmail, setRegisterEmail] = useState("");
+  const [logInPassword, setLogInPassword] = useState("");
+  const [registerPassword, setRegisterPassword] = useState("");
+  const [isSigningIn, setIsSigningIn] = useState(false);
+  const [isSigningUp, setIsSigningUp] = useState(false);
+  const [activeTab, setActiveTab] = useState("login");
+
+  const onSignIn = async (e) => {
+    e.preventDefault();
+    if (!isSigningIn) {
+      setIsSigningIn(true);
+      await doSignInWithEmailAndPassword(logInEmail, logInPassword).catch(
+        (err) => {
+          isSigningIn(false);
+        }
+      );
+    }
+  };
+
+  const onSignUp = async (e) => {
+    e.preventDefault();
+    if (!isSigningUp) {
+      setIsSigningUp(true);
+      await doCreateUserWithEmailAndPassword(
+        registerEmail,
+        registerPassword
+      ).catch((err) => {
+        setIsSigningUp(false);
+      });
+    }
+  };
+
+  const onGoogleSignIn = async () => {
+    if (!isSigningIn) {
+      setIsSigningIn(true);
+      await doSignInWithGoogle().catch((err) => {
+        setIsSigningIn(false);
+      });
+    }
+  };
+
+  const handleTabClick = (tabName) => {
+    if (tabName !== activeTab) {
+      setActiveTab(tabName);
+    }
+  };
+
+  return (
+    //if user logged in redirect to home - add here
+    <div className="container mt-5">
+      <ul className="nav nav-tabs">
+        <li className="nav-item">
+          <a
+            className={`nav-link ${activeTab === "login" ? "active" : ""}`}
+            onClick={() => handleTabClick("login")}
```

Comment:
Try looking at react router, you can routes for each of these pages instead of managing this way